src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -300,6 +300,30 @@ public Dataset getDatasetData(Program program, UUID experimentId, UUID datasetId
         log.debug("fetching dataset: " + datasetId + " for experiment: " + experimentId + ".  including stats: " + stats);
         log.debug("fetching observationUnits for dataset: " + datasetId);
         List<BrAPIObservationUnit> datasetOUs = ouDAO.getObservationUnitsForDataset(datasetId.toString(), program);
+
+        //Add years to the addition_info elements
+        Map<String, Integer> studyDbId2Year = new HashMap<>();  // used to prevent the same study from being fetched repeatedly.
```

Comment:
Maybe we should create a card to cache seasons in redis, I think this has come up elsewhere in the experiment processor as well. It looks like some of the study dao methods use the program cache but the one used here doesn't so makes sense for now but maybe in the future we could look at utilizing the redis cache.

Reply:
I created the story **[BI-2126](https://breedinginsight.atlassian.net/browse/BI-2261) Cache seasons in redis**.
The Code is now using _getStudyByEnvironmentId()_ to fetch the Study from cache.